File: api/tokenizer.js
function tokenize(data) {
  const sectionRegex = /\n?Keywords:|\n\n/g;
  const sections = data.split(sectionRegex).filter((element) => element);
  const paragraphs = [];

  if (sections.length == 8) {
    for (let i = 0; i < sections.length; i += 2) {
      const paragraph = sections[i].trim();
      const keywordsString = sections[i + 1].trim().slice(1, -1);
      const keywords = keywordsString.split(", ");

      paragraphs.push({ paragraph, keywords });
    }
  } else {
    for (let i = 0; i < 4; i++) {
      paragraphs.push({ paragraph: "", keywords: "" });
    }
  }

  return paragraphs;
}

module.exports = {
  tokenize: tokenize,
};

File: api/utils/tokenizer.py
import re

def tokenize(data):
    section_regex = r"\n?Keywords:|\n\n"
    sections = re.split(section_regex, data)
    sections = [section.strip() for section in sections if section.strip()]

    paragraphs = []

    if len(sections) == 8:
        for i in range(0, len(sections), 2):
            paragraph = sections[i].strip()
            keywords_string = sections[i + 1].strip()[1:-1]
            keywords = keywords_string.split(", ")
            paragraphs.append({"paragraph": paragraph, "keywords": keywords})
    else:
        for _ in range(4):
            paragraphs.append({"paragraph": "", "keywords": ""})

    return paragraphs

